azurerm/resource_arm_sql_server.go

```diff
@@ -88,19 +88,17 @@ func resourceArmSqlServerCreateUpdate(d *schema.ResourceData, meta interface{})
 		},
 	}
 
-	createResp, createErr := client.CreateOrUpdate(resGroup, name, parameters, make(chan struct{}))
-	resp := <-createResp
-	err := <-createErr
+	future, err := client.CreateOrUpdate(ctx, resGroup, name, parameters)
 	if err != nil {
-		// if the name is in-use, Azure returns a 409 "Unknown Service Error" which is a bad UX
-		if utils.ResponseWasConflict(resp.Response) {
-			return fmt.Errorf("SQL Server names need to be globally unique and '%s' is already in use.", name)
-		}
+		return err
```

Comment:
I assume this doesn't happen anymore? i.e. that confusing error was replaced?

Reply:
I couldn't see this field being returned anymore (but taking another look, I see it now) - I've pushed a commit to fix